inc/ctest.h

```diff
@@ -234,15 +235,18 @@ if (type##_Compare((A), (B))) \
 #endif
 
 #define CTEST_ASSERT_ARE_NOT_EQUAL_WITH_MSG(type, A, B, message) \
-if (!type##_Compare((A), (B))) \
-{ \
-    char actualString[1024]; \
-    type##_ToString(actualString, sizeof(actualString), (A)); \
-    printf("  Assert failed in line %d: Inequality expected. Got equality, value:%s. %s\n", __LINE__, actualString, (message)); \
-    if(g_CurrentTestFunction!=NULL) *g_CurrentTestFunction->TestResult = TEST_FAILED; \
-    do_jump(&g_ExceptionJump, (void*)"some expected string", actualString); \
-; \
-}
+do {                                                                                                                    \
+    char expectedString[1024];                                                                                          \
+    char actualString[1024];                                                                                            \
+    type##_ToString(expectedString, sizeof(expectedString), (A));                                                       \
+    type##_ToString(actualString, sizeof(actualString), (B));                                                           \
+    if (strcmp(expectedString, actualString) == 0)                                                                      \
```

Comment:
>strcmp [](start = 8, length = 6)

I actually do not like the comparison based on the string value. The reason is that the stringify function might for example cut out a part of a long array for brevity. Or to make comparing two values accurate maybe more than 1024 bytes are needed. But this makes it better than before I guess.

Btw have you tried assigning the expected and actual.

like:

```c
type A_value = (A);
type B_value = (B);

if (!type##_Compare(A_value, B_value)
...

```